src/Prism.Plugin.Popups/Dialogs/PopupDialogService.cs

```diff
@@ -49,6 +51,16 @@ public void ShowDialog(string name, IDialogParameters parameters, Action<IDialog
                     }
                 }
 
+                if (!parameters.TryGetValue<bool>(KnownPopupDialogParameters.Animated, out var animated))
+                    animated = true;
+
+                if (!parameters.TryGetValue<bool>(KnownPopupDialogParameters.IsAnimationEnabled, out var isAnimationEnabled))
+                {
+                    var popupDialogLayoutIsAnimationEnabled = PopupDialogLayout.GetIsAnimationEnabled(view);
+                    isAnimationEnabled = popupDialogLayoutIsAnimationEnabled ?? true;
+                }
+                popupPage.IsAnimationEnabled = isAnimationEnabled;
```

Comment:
to look at implementation from the KnownParameters... this just generally makes no sense.

Why would you want to have an attached property for the Dialog to set a true/false on IsAnimationEnabled when there is already a property IsAnimationEnabled on the PopupPage? This then has no effect what so ever on the animated value we pass. If `popupPage.IsAnimationEnabled == false` then it shouldn't matter whether we pass true or false to the PushPopupPage... that said there is the question what should happen if IsAnimationEnabled is false, and someone explicitly specifies that they want animated to be true, I would imagine that case we would want to override the set value to true even though the page has IsAnimationEnabled set to false

Reply:
> Why would you want to have an attached property for the Dialog to set a true/false on IsAnimationEnabled when there is already a property IsAnimationEnabled on the PopupPage?

A user of IDialogService has no access to the PopupPage and therefore cannot set the IsAnimationEnabled property. The attached property provides a way to do this in the XAML of the dialog. The KnownPopupDialogParameters.IsAnimationEnabled provides a way to do this when calling ShowDialog. These are two different approaches, which I adapted from KnownDialogParameters.CloseOnBackgroundTapped.

> If popupPage.IsAnimationEnabled == false then it shouldn't matter whether we pass true or false to the PushPopupPage... that said there is the question what should happen if IsAnimationEnabled is false, and someone explicitly specifies that they want animated to be true, I would imagine that case we would want to override the set value to true even though the page has IsAnimationEnabled set to false

With Rg.Plugins.Popup setting `PopupPage.IsAnimationEnabled == false` completely disables popup animations, regardless of what is passed for `animated` parameter into PushPopupPage.

Reply:
ok I see your issue there. The attached property that you've added should be the only way to set that. That should not also be a NavigationParameter. Essentially if someone wants to explicitly disable Animation on a Dialog they should use the attached property. If they want to control it call by call they should use the Navigation parameter `animated`

Reply:
Exactly.
To confirm the todo:

- [ ] Remove `KnownPopupDialogParameters.IsAnimationEnabled` and corresponding logic